### PR TITLE
Send stacktrace details (fileName/methodName/lineNumber) in PG error response

### PR DESCRIPTION
**Pull request by matriv**
@mfussenegger Please review
Please suggest if we want more "sophisticated" analysis of the stacktrace elements...

### CODE REVIEW

File: sql/src/main/java/io/crate/protocols/postgres/Messages.java [hunk 1]
```diff
@@ -200,6 +200,20 @@ static void sendErrorResponse(Channel channel, Throwable throwable) {
         final String message = Exceptions.messageOf(throwable);
         byte[] msg = message.getBytes(StandardCharsets.UTF_8);
         byte[] severity = "ERROR".getBytes(StandardCharsets.UTF_8);
+        byte[] lineNumber = null;
+        byte[] fileName = null;
+        byte[] methodName = null;
+
+        if (throwable.getStackTrace() != null) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/protocols/postgres/Messages.java]: Please assign the getStackTrace result to a local variable. Although it is creating the stacktrace only once it's still synchronized and cloning the thing.

File: sql/src/test/java/io/crate/integrationtests/PostgresITest.java [hunk 2]
```diff
@@ -514,6 +515,20 @@ public void testErrorRecoveryFromErrorsOutsideSqlOperations() throws Exception {
...
 
+    @Test
+    public void testErrorDetailsFromStackTraceInErrorResponse() throws Exception {
+        try (Connection conn = DriverManager.getConnection(JDBC_POSTGRESQL_URL, properties)) {
+            conn.setAutoCommit(true);
+            conn.createStatement().executeUpdate("create table t (i integer)");
+            ensureYellow();
+            conn.createStatement().executeUpdate("insert into t (i) values ('abc')");
+        } catch (PSQLException e) {
+            assertThat(e.getServerErrorMessage().getFile(), is("ValueNormalizer.java"));
```
Comment (mfussenegger) [inline sql/src/test/java/io/crate/integrationtests/PostgresITest.java]: imho it's enough to check for notNull. This test shouldn't care about where the error is thrown but just verify that the information is there.

**Comment by matriv** [comment]
@mfussenegger plz check fixup

File: sql/src/main/java/io/crate/protocols/postgres/Messages.java [hunk 3]
```diff
@@ -204,8 +204,9 @@ static void sendErrorResponse(Channel channel, Throwable throwable) {
         byte[] fileName = null;
         byte[] methodName = null;
 
-        if (throwable.getStackTrace() != null) {
-            StackTraceElement stackTraceElement = throwable.getStackTrace()[0];
+        StackTraceElement[] stackTrace = throwable.getStackTrace();
+        if (stackTrace != null) {
```
Comment (mfussenegger) [inline sql/src/main/java/io/crate/protocols/postgres/Messages.java]: what if the length is zero?

**Comment by matriv** [comment]
@mfussenegger check fixup please

**Pull request merged**